src/openrct2/network/Network.cpp

```diff
@@ -2532,40 +2539,64 @@ void Network::Server_Handle_TOKEN(NetworkConnection& connection, [[maybe_unused]
 void Network::Client_Handle_OBJECTS_LIST(NetworkConnection& connection, NetworkPacket& packet)
 {
     auto& repo = GetContext()->GetObjectRepository();
-    uint32_t size;
-    packet >> size;
-    log_verbose("client received object list, it has %u entries", size);
-    if (size > OBJECT_ENTRY_COUNT)
+
+    uint32_t index = 0;
+    uint32_t totalObjects = 0;
+    packet >> index >> totalObjects;
+
+    if (index == 0)
+    {
+        // Start of objects list.
```

Comment:
I wonder if we should make
```c++
if (index == OBJECT_LIST_START)
{
    _missingObjects.clear();
}
```

Where `OBJECT_LIST_START` is `0`

Reply:
I've changed it to send a bool if its the start of the list, this should clarify the purpose a bit better and the index is obsolete since I can use the count of _missingObjects.